src/protocols/bdx/AsyncTransferFacilitator.cpp

```diff
@@ -170,21 +181,20 @@ void AsyncResponder::NotifyEventHandled(const TransferSession::OutputEventType e
     ChipLogDetail(BDX, "NotifyEventHandled : Event %s Error %" CHIP_ERROR_FORMAT,
                   TransferSession::OutputEvent::TypeToString(eventType), status.Format());
 
-    // If this is the end of the transfer (whether a clean end, or some sort of error condition), ensure that
-    // we destroy ourselves after processing any output events that might have been generated by
-    // the transfer session to handle end-of-transfer (e.g. in some error conditions it might want to send
-    // out a status report).
+    // If this is the end of the transfer (whether a clean end, or some sort of error condition), ensure
+    // that we destroy ourselves after unwinding the processing loop in the ProcessOutputEvents API.
+    // We can ignore the status for these messages since the state machine is in a bad, unrecoverable
+    // state and we should stop processing events and clean up.
```

Comment:
```suggestion
    // We can ignore the status for output events because none of them are supposed to result in
    // us sending a StatusReport, and that's all we use the status for.
    //
    // In particular, for kTransferTimeout, kAckEOFReceived, and kStatusReceived per spec we
    // are not supposed to reply with a StatusReport.  And for kInternalError the state machine
    // is in an unrecoverable state of some sort, and we should stop trying to make use of it.
```